platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/AddressBookUtils.java

```diff
@@ -16,46 +16,97 @@
 
 package com.swirlds.common.system.address;
 
+import static com.swirlds.common.system.address.Address.ipString;
+
+import com.swirlds.common.formatting.TextTable;
 import com.swirlds.common.system.NodeId;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import java.net.InetAddress;
 import java.net.UnknownHostException;
 import java.text.ParseException;
 import java.util.Objects;
-import java.util.function.Function;
 
 /**
  * A utility class for AddressBook functionality.
  */
 public class AddressBookUtils {
+
     private AddressBookUtils() {}
 
     /**
-     * Parses an address from a single line of text.  The address must be in the form used in config.txt
-     * <p>
-     * If the memo field can be parsed, the provided memo field is ignored.  If there is no memo field parsed, the
-     * provided memo field is used.
+     * Serializes an AddressBook to text in the form used by config.txt.
      *
-     * @param addressLine         the string to parse an Address form.
-     * @param id                  the id to give to the parsed Address.
-     * @param isOwnHostDeterminer a function to determine if isOwn should be true given an InetAddress.
-     * @param memo                the memo text for the address.
-     * @return the Address parsed from the addressLine.
-     * @throws ParseException if there is any problem with creating an Address from the addressLine.
+     * @param addressBook the address book to serialize.
+     * @return the config.txt compatible text representation of the address book.
      */
     @NonNull
-    public static Address parseAddressConfigText(
-            @NonNull final String addressLine,
-            @NonNull final NodeId id,
-            @NonNull final Function<InetAddress, Boolean> isOwnHostDeterminer,
-            @NonNull final String memo)
-            throws ParseException {
-        Objects.requireNonNull(addressLine, "The addressLine must not be null.");
-        Objects.requireNonNull(id, "The id must not be null.");
-        Objects.requireNonNull(isOwnHostDeterminer, "The isOwnHostDeterminer must not be null.");
-        Objects.requireNonNull(memo, "The memo must not be null.");
-        final String[] parts = addressLine.trim().split(",");
-        if (parts.length < 8 || parts.length > 9) {
+    public static String addressBookConfigText(@NonNull final AddressBook addressBook) {
+        Objects.requireNonNull(addressBook, "The addressBook must not be null.");
+        final TextTable table = new TextTable().setBordersEnabled(false);
+        for (final Address address : addressBook) {
+            final String memo = address.getMemo();
+            final boolean hasMemo = !memo.trim().isEmpty();
+            final boolean hasInternalIpv4 = address.getAddressInternalIpv4() != null;
+            final boolean hasExternalIpv4 = address.getAddressExternalIpv4() != null;
+            table.addRow(
+                    "address,",
```

Comment:
No idea about the internal of `TextTable` but it looks to me that `TextTable` should add the "," instead of doing it for each column by hand.

Reply:
I agree, Textable should be configurable with whether or not to add the comma.   I'll look into how hard it would be to add it.

Reply:
I created an issue for this as it is not part of the scope of this PR: https://github.com/hashgraph/hedera-services/issues/7267